### PR TITLE
Login/register css

### PR DESCRIPTION
**Pull request by calmoon808**
Seperated registration survey in to parts, added CSS, changed how password was getting passed to avoid a google security error I was getting.

### CODE REVIEW

File: src/components/LoginComponent/LoginComponent.jsx [hunk 1]
```diff
@@ -48,11 +48,15 @@ class LoginComponent extends Component {
   }
 
   render() {
+    if (this.props.loginError === true){
+      alert("Email and password combination to not match.");
```
Comment (danieltichiyama) [inline src/components/LoginComponent/LoginComponent.jsx]: "to" to "do"

File: src/components/RegisterComponent/RegisterComponent.jsx [hunk 2]
```diff
@@ -51,7 +51,12 @@ class RegisterComponent extends Component {
       alert("Password must be at least 6 characters long.");
       return false;
     };
-    return this.props.takingSurvey(this.state);
+    
+    await this.props.dispatchBcrypt({ password: this.state.password });
+    return this.setState({ password: this.props.password}, () => {
+      delete this.state.confirm_password;
+      this.props.takingSurvey(this.state);
+    })
```
Comment (danieltichiyama) [inline src/components/RegisterComponent/RegisterComponent.jsx]: does this have to go into a .then() statement???

File: src/index.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ const enhancer = composeEnhancers(applyMiddleware(ReduxThunk)); //applyMiddlewar
 const store = createStore(reducer, enhancer);
 
 ReactDOM.render(
-  <Provider store={store}>
+  <Provider store={store}z>
```
Comment (danieltichiyama) [inline src/index.js:25]: delete z

File: src/reducers/index.js [hunk 4]
```diff
@@ -175,6 +185,9 @@ let reducer = (store = initialStore, action) => {
     case IMAGE_UPLOAD:
       return Object.assign({}, store, { images: action.payload }); 
       
+    case BCRYPT:
+      return Object.assign('', store, { bcrypt: action.payload })
```
Comment (danieltichiyama) [inline src/reducers/index.js]: change from '' to {}

File: src/components/RegisterSurveyComponent/RegisterSurveyComponent.jsx [hunk 5]
```diff
@@ -46,243 +82,395 @@ class RegisterSurveyComponent extends Component {
...
+              <input 
+                id="veryActive"
+                type="radio"
+                name="activityLevel"
+                value="4"
+                onClick={(e) => {
+                  this.setState({ activity_level_id: e.target.value });
+                }}
+              />
+              <label htmlFor="veryActive">veryActive</label>
```
Comment (jkhirano) [inline src/components/RegisterSurveyComponent/RegisterSurveyComponent.jsx]: Change to: <label htmlFor="veryActive">Very Active</label>

File: src/components/RegisterSurveyComponent/RegisterSurveyComponent.jsx [hunk 6]
```diff
@@ -46,243 +82,395 @@ class RegisterSurveyComponent extends Component {
...
+            <button className={styles.registerButton} onClick={this.handleNext2}>
+              Next
+            </button>
+          </>
+        ) : ('')}
+
+        {this.state.goalSelect ? (
+          <>
+            <h2>Goal</h2>
+            <li>
```
Comment (jkhirano) [inline src/components/RegisterSurveyComponent/RegisterSurveyComponent.jsx]: The list/button is getting cut off by the navbar in some views (galaxy / iphone). I think you need to put the goal list elements in a div and then apply flex to fix it so it's responsive.

Comment (jkhirano) [reply src/components/RegisterSurveyComponent/RegisterSurveyComponent.jsx]: Flex box should be applied to all other list divs as well

File: src/components/LoginComponent/LoginComponent.jsx (hunk 1 above)
Comment (jkhirano) [inline src/components/LoginComponent/LoginComponent.jsx]: Alert initially works when my password is wrong. But as I try to delete so I can re-enter it, the alert keeps popping up with every delete key pressed. The very last letter doesn't delete so I can't re-enter password.

**Review by jkhirano — COMMENTED** [review]
css looks good :) only have a few things to change..I'm not sure if fixing the password popup is necessary since we're short on time. If it's not fixed, we'll just have to make sure that doesn't show up during the presentation.

**Comment by calmoon808** [comment]
K everything rerouted now. changed the register bcrypt process back to its original state. register survey component is now part of register component. fixed the delete bug after failed login attempt. Hopefully nothing else comes up

File: src/actions/index.js [hunk 7]
```diff
@@ -130,17 +131,19 @@ export const actionsLoadActivity = () => async dispatch => {
...
         payload: response.data
       });
+      return true
     })
     .catch(err => {
-      console.log("Error in actionsLoginsubmit: ", err);
+      dispatch({
+        type: LOGIN_ERROR
+      });
+      return false
```
Comment (danieltichiyama) [inline src/actions/index.js:146]: should it return false here or should it return the dispatch?  also, maybe we ought to console.log the error like the other actions?

File: src/components/LoginComponent/LoginComponent.jsx [hunk 8]
```diff
@@ -33,9 +32,15 @@ class LoginComponent extends Component {
...
       return this.setState({ error: true });
     } else {
-      return this.props.dispatchLoginSubmit(this.state);
+      await this.props.dispatchLoginSubmit(this.state);
+      return this.setState({ error: false }, () => {
+        if(this.props.loginError){
+          alert("Email and password combination to not match.");
+          return false;
+        }
+      })
```
Comment (danieltichiyama) [inline src/components/LoginComponent/LoginComponent.jsx]: should this be a .then??

File: src/components/RegisterComponent/RegisterComponent.module.scss [hunk 9]
```diff
@@ -1,33 +1,47 @@
+$turquoise: #04c9b5;
+$red: #fc4a1a;
```
Comment (danieltichiyama) [inline src/components/RegisterComponent/RegisterComponent.module.scss:2]: oh, shit, nice.

File: src/index.js (hunk 3 above)
Comment (danieltichiyama) [reply src/index.js:25]: please delete z

**Review by danieltichiyama — APPROVED** [review]

**Pull request merged**